src/components/ChartMenu/ChartMenu.tsx

```diff
@@ -0,0 +1,124 @@
+import styles from "@components/ChartMenu/ChartMenu.module.scss";
+import { Chart } from "@components/ChartMenu/Chart/Chart";
+import { DragEvent, useState, useEffect, useRef } from "react";
+import { ChartElement, LineFigure } from "@components/ChartMenu/ChartElement";
+import { selectMeasurementByName } from "@models/PodData/PodData";
+export const ChartMenu = () => {
+  let boards = useSelector((state: RootState) => state.podData.boards);
+  let [chartElements, setChartElements] = useState([] as ChartElement[]);
+  let chartIndex = useRef(0);
+
+  useEffect(() => {
+    setOrderedChartElements((prevChartElements) => {
+      return prevChartElements.map((chartElement) => {
+        let newLines = chartElement.lines.map((line) => {
+          let newValue = selectMeasurementByName(boards, line.name)
+            .value as number;
+          let newVector = [...line.vector, newValue];
+          if (newVector.length > 200) {
+            newVector = newVector.slice(newVector.length - 200);
+          }
+          return {
+            name: line.name,
+            vector: newVector,
+            color: line.color,
+          };
+        });
+        return { id: chartElement.id, lines: newLines };
+      });
+    });
+  }, [boards]);
+
+  function setOrderedChartElements(
+    callback: (prevValues: ChartElement[]) => ChartElement[]
+  ): void {
+    setChartElements((prevValues) => {
+      return callback(prevValues).sort((first, second) => first.id - second.id);
+    });
+  }
+
+  function addChartElement(element: ChartElement) {
+    setOrderedChartElements((prevValues) => {
+      return [...prevValues, element];
+    });
+  }
+
+  function handleDrop(ev: DragEvent<HTMLDivElement>) {
+    addChartElement({
+      id: chartIndex.current,
+      lines: [
+        {
+          name: ev.dataTransfer.getData("text/plain"),
+          vector: [],
+          color: { h: 32, s: 60, l: 80 },
+        },
+      ],
+    });
+    chartIndex.current++;
+  }
+
+  function handleDropOnChart(id: number, measurementName: string) {
+    let chartElement = chartElements.find((element) => element.id == id)!;
+    if (isAlreadyInChart(chartElement, measurementName)) {
+      return;
+    }
+    let newH = (32 + chartElement.lines.length * 123) % 360;
+    let newChartElement = {
+      ...chartElement,
+      lines: [
+        ...chartElement.lines,
+        {
+          name: measurementName,
+          vector: [0],
+          color: {
+            h: newH,
+            s: 60,
+            l: 80,
+          },
+        },
+      ],
+    } as ChartElement;
+    let restOfElements = chartElements.filter((element) => element.id != id);
+    setOrderedChartElements(() => {
+      return [...restOfElements, newChartElement];
+    });
+  }
+
+  function isAlreadyInChart(
+    chartElement: ChartElement,
+    measurementName: string
+  ): boolean {
+    return (
+      chartElement.lines.findIndex((line) => line.name == measurementName) != -1
+    );
+  }
+
```

Comment:
I sometimes perform functions outside the main function of the file (at the end of the file), I see that you always do it inside the function and before the return. Should it always be like that?

Reply:
Functions outside of the component are global to all the instances of the component. That means that you cant use data of a component inside of the function directly, it has to be passed as parameters. The benefit of defining them outisde its that they are not redeclared every time the component renders. Maybe I should look into doing it outside like you do, unless I dont have any other alternative.